Emby.Server.Implementations/Updates/InstallationManager.cs

```diff
@@ -107,205 +107,98 @@ public class InstallationManager : IInstallationManager
-        /// <param name="classification">The classification.</param>
-        /// <returns>PackageVersionInfo.</returns>
-        public PackageVersionInfo GetLatestCompatibleVersion(IEnumerable<PackageInfo> availablePackages, string name, string guid, Version currentServerVersion, PackageVersionClass classification = PackageVersionClass.Release)
-        {
-            var package = availablePackages.FirstOrDefault(p => string.Equals(p.guid, guid ?? "none", StringComparison.OrdinalIgnoreCase))
-                            ?? availablePackages.FirstOrDefault(p => p.name.Equals(name, StringComparison.OrdinalIgnoreCase));
-
-            return package?.versions
-                .OrderByDescending(x => x.Version)
-                .FirstOrDefault(v => v.classification <= classification && IsPackageVersionUpToDate(v, currentServerVersion));
+            return GetCompatibleVersions(
+                package.versions,
+                minVersion,
+                classification);
         }
 
-        /// <summary>
-        /// Gets the available plugin updates.
-        /// </summary>
-        /// <param name="applicationVersion">The current server version.</param>
-        /// <param name="withAutoUpdateEnabled">if set to <c>true</c> [with auto update enabled].</param>
-        /// <param name="cancellationToken">The cancellation token.</param>
-        /// <returns>Task{IEnumerable{PackageVersionInfo}}.</returns>
-        public async Task<IEnumerable<PackageVersionInfo>> GetAvailablePluginUpdates(Version applicationVersion, bool withAutoUpdateEnabled, CancellationToken cancellationToken)
+        /// <inheritdoc />
+        public async Task<IEnumerable<PackageVersionInfo>> GetAvailablePluginUpdates(CancellationToken cancellationToken = default)
         {
-            var catalog = await GetAvailablePackagesWithoutRegistrationInfo(cancellationToken).ConfigureAwait(false);
+            var catalog = await GetAvailablePackages(cancellationToken).ConfigureAwait(false);
 
             var systemUpdateLevel = _applicationHost.SystemUpdateLevel;
 
             // Figure out what needs to be installed
-            return _applicationHost.Plugins.Select(p =>
+            return _applicationHost.Plugins.Select(x =>
             {
-                var latestPluginInfo = GetLatestCompatibleVersion(catalog, p.Name, p.Id.ToString(), applicationVersion, systemUpdateLevel);
-
-                return latestPluginInfo != null && latestPluginInfo.Version > p.Version ? latestPluginInfo : null;
-            }).Where(i => i != null)
-            .Where(p => !string.IsNullOrEmpty(p.sourceUrl) && !CompletedInstallations.Any(i => string.Equals(i.AssemblyGuid, p.guid, StringComparison.OrdinalIgnoreCase)));
+                var compatibleversions = GetCompatibleVersions(catalog, x.Name, x.Id, x.Version, systemUpdateLevel);
+                return compatibleversions.FirstOrDefault(y => y.Version > x.Version);
```

Comment:
this doesn't feel right - what if there are several versions? shouldn't we be taking maximum available version? I don't see sorting anywhere

Reply:
They are already sorted in the json file.

Reply:
Then they should conform to our standard. And since it’s all automated it should always stay the same.